Update README.md (#50)
Aam-Digital/aam-services: docs/developer/README.md

You can open the `http://` version directly again. 

# Project Setup Guide

This guide includes instructions for both a **full local setup with Docker** and a **minimal setup without Docker**.

---

## Full Local Setup (with Docker)

### Step 1: Start Docker Services
1. Run the `docker-compose` file to start all related services:
```shell
   docker-compose up
```
2. If needed, switch the image in `docker-compose.yml` from `aam-sqs-mac` to `aam-sqs-linux` for compatibility.

### Step 2: Configure Keycloak

> 3. Open the Keycloak Admin UI at [http://localhost:8080](http://localhost:8080) with the following credentials.

Reviewer: ```suggestion
3. Open the Keycloak Admin UI at [http://localhost:8080](http://localhost:8080) with the credentials defined in the docker-compose file. Defaults are:
```

```
username - admin
password - docker
```
4. Create a new realm called **dummy-realm** by importing the [realm base configuration file](https://github.com/Aam-Digital/ndb-setup/blob/master/baseConfigs/realm_config.example.json).
5. Under **Keycloak Realm > Clients** ([http://localhost:8080/admin/master/console/#/dummy-realm/clients](http://localhost:8080/admin/master/console/#/dummy-realm/clients)), import the client configuration using the [client config file](https://github.com/Aam-Digital/ndb-setup/blob/master/keycloak/client_config.json).

> 6. In the new realm, create a user and assign relevant roles.

Reviewer: ```suggestion
6. In the new realm, create a user and assign relevant roles. (Usually you will want at least "user_app" and/or "admin_app" role to be able to load the basic app config.)
```

### Step 3: Set Up CouchDB
7. Access CouchDB at [http://localhost:5984/_utils/#database/app/_all_docs](http://localhost:5984/_utils/#database/app/_all_docs).
8. Create a new database name as `app`.
9. Add a document of type **Config:CONFIG_ENTITY** to the `app` database (e.g., from [dev.aam-digital.net CouchDB instance](https://dev.aam-digital.net/db/couchdb/_utils/#database/app/Config%3ACONFIG_ENTITY)).

> #### Note: If you get an error while adding a document (eg. document update conflict warning) remove the "_rev": "value".

Reviewer: ```suggestion
**Note: If you get an error while adding a document (eg. document update conflict warning) remove the "_rev": "value".**
```

### Step 4: Start the Backend

> 9. Retrieve the `public_key` for **dummy-realm** from [http://localhost:8080/realms/dummy-realm](http://localhost:8080/realms/dummy-realm) and add it to the `.env` file for the replication backend as `JWT_PUBLIC_KEY`.

Reviewer: ```suggestion
9. Clone the [replication-backend](https://github.com/Aam-Digital/replication-backend) repository (if you do not already have it available locally) and follow the setup instructions of its README to install dependencies.
9. Retrieve the `public_key` for **dummy-realm** from [http://localhost:8080/realms/dummy-realm](http://localhost:8080/realms/dummy-realm) and add it to the `.env` file for the replication backend as `JWT_PUBLIC_KEY`.
```

10. Start the replication backend:
```shell
    npm run start:dev
```

### Step 5: Start the Frontend

> 11. Update `environment.ts` or `assets/config.json` with the following settings:

Reviewer: ```suggestion
11. Update `environment.ts` or `assets/config.json` with the following settings, in order to run the app in "synced" mode using the backend services:
```

- Replication Backend: [http://localhost:3000](http://localhost:3000)
- Keycloak: [http://localhost:8080](http://localhost:8080)
- CouchDB: [http://localhost:5984](http://localhost:5984)

---

## Minimal Setup (Without Docker)

For a basic setup without Docker, follow these steps.

### Step 1: Set Up CouchDB
1. Install CouchDB on Mac:
```shell
   brew install couchdb
```
2. Configure CouchDB:
```shell
   nano /opt/homebrew/etc/local.ini
```
3. Restart CouchDB:
```shell
   brew services restart couchdb
```
4. Access CouchDB at [http://localhost:5984/_utils/#](http://localhost:5984/_utils/#).

### Step 2: Create Databases
1. Create the following databases in CouchDB:
- `app`
- `app-attachments`
- `notification-webhook`
- `Report-calculation`

3. In the `app` database, add these configuration documents:
- **Config:CONFIG_ENTITY**
- **Config:Permissions**

**Note**: Please reach out to the tech team to get the specific details for these configuration documents.

### Step 3: Set Up Replication Backend
- Follow the project setup instructions provided in the [Backend Setup Guideline](https://github.com/Aam-Digital/replication-backend/blob/master/README.md).

> Once completed, you’ll have a minimal local environment without Docker.

Reviewer: Does this one actually work, @Abhinegi2 ? Or does this run into the issue of data not syncing to the local CouchDB because authentication is missing?

Author: @sleidig  It works, but I think anyone working in the app needs to sync with the database at least once, so this might not be very helpful. Can I remove this? However, for just fetching the data—like we do in public forms—it can still retrieve the data successfully.

Reviewer: if the normal create/update entities is not supported, then I think this option is more confusing than helpful. Let's remove it if that is the only case
